### ISSUE TITLE
Wrong error message (out parameter must be assigned to before control leaves)

### ISSUE THREAD

**Issue by OwnageIsMagic** [dotnet/roslyn#68380]
**Version Used**: 8.0.100-preview.4.23260.5
**Steps to Reproduce**:
```c#
void f(out int i)
{
    int q = 0;
    i = ref q;
}
```
> error CS0177: The out parameter 'i' must be assigned to before control leaves the current method

https://sharplab.io/#v2:EYLgtghglgdgPgAQCwAIBmAKA9gVwC4qwFQCUAsAFADelKdhMBAjigLwoAMA3LfVGygBOAUzQomPCgF8gA==

**Diagnostic Id**: `CS0177`

**Expected Behavior**: Appropriate message.
